### ISSUE TITLE
[indent] Typescript experimental decorator on class property

### ISSUE THREAD

**Issue by Tallyrald** [typescript-eslint/typescript-eslint#4032]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JS
{
	'env': {
		'browser': true,
		'es6': true,
		'node': true,
	},
	'parser': '@typescript-eslint/parser',
	'parserOptions': {
		'project': ['./tsconfig.json'],
		'sourceType': 'module',
	},
	'plugins': [
		'@typescript-eslint',
	],
	'rules': {
		'indent': ['error', 'tab', {'SwitchCase': 1}],
	},
	'settings': {
		'import/parsers': {
		  '@typescript-eslint/parser': ['.ts', '.tsx'],
		},
		'import/resolver': {
			'typescript': {
				'directory': './tsconfig.json',
			},
		},
	},
}
```

```TS
class DemonstrationClass {

	@Column({
		default: 0,
	})
	amount: number;
	/* ^^^^ Expected indentation of 2 tabs but found 1. */
}
```

tsconfig.json
```json
{
	"compilerOptions": {
		"allowSyntheticDefaultImports": true,
		"baseUrl": "./ts",
		"emitDecoratorMetadata": true,
		"esModuleInterop": true,
		"experimentalDecorators": true,
		"module": "commonjs",
		"moduleResolution": "node",
		"noImplicitAny": true,
		"outDir": "built",
		"removeComments": true,
		"sourceMap": true,
		"target": "es2020",
	},
	"exclude": [],
	"include": ["ts"]
}
```

**Expected Result**

No linting errors.

**Actual Result**

Indent rule reports "Expected indentation of 2 tabs but found 1." - see the class example above.

**Additional Info**

No exceptions from ESLint are reported.
The exact same rules work fine (no linting errors) on ESLint 7.32.0. I've already opened an issue there, but I was redirected here. See https://github.com/eslint/eslint/issues/15182

Any help is appreciated.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.0.1` |
| `node`                             | `16.11.1` |

**Comment by itpropro** [comment]
Many people don't want to use prettier, as it's way too opinionated and use pure eslint instead.
I don't know if saying that the current implementation is and will stay broken and leaving it in the codebase makes any sense. Either fix or remove it, otherwise people will run into this issue and opening new GitHub issues as long as it stays broken...

**Comment by bradzacher** [comment]
> fix it

That's what the issue is there for. We're waiting for a motivated community member or members to improve it. Nobody has been willing this far.

If this is a problem that you run into we'll very gladly accept a PR!

> remove it

It has some support and does better than the base rule, but it has a lot of issues.

If we delete it, then people will complain to ESLint core that it doesn't support TS, and they will then redirect them to us. It will create support work across more repos if we delete it.
Removing it is not a good idea.

---

Instead I'm much happier being open and transparent with everyone.

I'm more than happy to point people at the meta issue when a new issue is opened in error. The current rate of about one per month is quick and easy to handle.

**Comment by itpropro** [comment]
I get your point. I would personally rather not release something half broken and tell the people to fix it themselves then release it somewhat working, But it seems like its better to have this than nothing. Although, for me there is no difference in using the eslint "indent" rule vs. using the "@typescript-eslint/indent" rule.

**Comment by bradzacher** [comment]
> I would personally rather not release something half broken and tell the people to fix it themselves then release it somewhat working

When it was first released (over 3 years ago), it was working for the cases it was defined for. Back then the plugin had a tens of thousands of weekly downloads and it worked fine.
But over time as more and more people have onboarded onto ESLint off of TSLint (we're now at 12 MILLION weekly downloads) - more and more code has been linted with it, and more and more cases and combinations have been found. The combinatorial explosion has shown that the initial approach doesn't work and it needs reworking.

Reworking it takes time and effort. Two things in short supply in a project run entirely by volunteers.
